### PR TITLE
[DO_NOT_MERGE] replacing @shopify/react-apollo with react-apollo

### PR DESCRIPTION
**Pull request by patsissons**
`@shopify/jest-mock-apollo` should match `web` once it switches to `react-apollo`.

### CODE REVIEW

File: packages/jest-mock-apollo/src/test/index.test.tsx [hunk 1]
```diff
@@ -40,6 +41,9 @@ describe('jest-mock-apollo', () => {
       context: {
         client: createGraphQLClient(),
       },
+      childContextTypes: {
+        client: PropTypes.object,
```
Comment (patsissons) [inline packages/jest-mock-apollo/src/test/index.test.tsx]: this was required to get the context pushed into mounted components.

Comment (patsissons) [reply packages/jest-mock-apollo/src/test/index.test.tsx]: @lemonmade do you have insight on this? Feels like we should probably abstract away the requirement to specify the `childContextTypes` when mounting.

Comment (lemonmade) [reply packages/jest-mock-apollo/src/test/index.test.tsx]: I'm surprised this didn't come up before. We usually create a custom mount function that abstracts this away (as in `mountWithAppContext` for web)

Comment (patsissons) [reply packages/jest-mock-apollo/src/test/index.test.tsx]: i suspect our internal `react-apollo` was wrapping with child context for us previously.

File: packages/react-i18n/src/decorator.tsx [hunk 2]
```diff
@@ -138,7 +138,7 @@ export function withI18n({id, fallback, translations}: WithI18nOptions = {}) {
       WithTranslation,
       WrappedComponent as React.ComponentClass<any>,
     );
-    return FinalComponent as React.ComponentClass<any> & C;
+    return FinalComponent as any;
```
[the remark this answers is not in the document]
Comment (patsissons) [reply packages/react-i18n/src/decorator.tsx]: this was no longer necessary, we can properly type the composed component now.

**Comment by patsissons** [comment]
I just rebased this to see if there was anything to keep, it looks like just some broken upgrades. I think i will probably close this PR out if no objections.

**Pull request closed without merging**